ext/afform/core/ang/af/afForm.component.js

```diff
@@ -302,7 +302,12 @@
           dialog.dialog('close');
         }
 
-        else if (metaData.redirect) {
+        else if (metaData.confirmation_type === 'show_confirmation_message') {
+          $scope.$parent.$parent.$parent.afformConfirmationMessage = metaData.confirmation_message;
+          $scope.$parent.$parent.$parent.afformDisplayForm = false;
```

Comment:
@kurund while it might not be so-very-Angular to do it this way, IMO the self-contained way to make this feature compatible with non-full-page afforms would be to change this code to something like
```
$element.hide();
const $confirmation = $('<div class="afform-confirmation" />');
$confirmation.text(metaData.confirmation_message);
$confirmation.insertAfter($element);
```